src/elements/content-sidebar/__tests__/DocgenSidebar.test.tsx

```diff
@@ -30,101 +28,63 @@ const defaultProps = {
     });
 
     test('should render DocGen sidebar component correctly with tags list', async () => {
-        let wrapper;
-        await act(async () => {
-            wrapper = getWrapper(defaultProps);
-        });
-        wrapper!.update();
-        const tagList = wrapper!.find('span.bcs-DocGen-tagPath');
-        expect(tagList).toHaveLength(26);
-        expect(wrapper).toMatchSnapshot();
+        renderComponent();
+        const tagList = await screen.findAllByTestId('bcs-TagsSection');
+        expect(tagList).toHaveLength(2);
     });
 
     test('should render empty state when there are no tags', async () => {
-        let wrapper;
-        await act(async () => {
-            wrapper = await getWrapper({
-                ...defaultProps,
-                getDocGenTags: noTagsMock,
-            });
+        renderComponent({
+            ...defaultProps,
+            getDocGenTags: noTagsMock,
         });
 
-        wrapper!.update();
-        const emptyState = wrapper!.find(FormattedMessage).at(0);
-        expect(emptyState.prop('defaultMessage')).toEqual('This document has no tags');
-        expect(emptyState).toHaveLength(1);
-        expect(wrapper).toMatchSnapshot();
+        const emptyState = await screen.findByText('This document has no tags');
+        expect(emptyState).toBeInTheDocument();
     });
 
     test('should render loading state', async () => {
-        const wrapper = await getWrapper({
+        renderComponent({
             ...defaultProps,
             getDocGenTags: noTagsMock,
         });
-        const loadingState = wrapper!.find(LoadingIndicator);
-        expect(loadingState).toHaveLength(1);
-        expect(wrapper).toMatchSnapshot();
+
+        const loadingState = await screen.getByRole('status'); // Assuming LoadingIndicator has a test id
+        expect(loadingState).toBeInTheDocument();
     });
 
     test('should re-trigger getDocGenTags on click on refresh button', async () => {
-        let wrapper;
-        await act(async () => {
-            wrapper = await getWrapper({
-                ...defaultProps,
-                getDocGenTags: errorTagsMock,
-            });
+        renderComponent({
+            ...defaultProps,
+            getDocGenTags: errorTagsMock,
         });
-        wrapper!.update();
 
-        const refreshBtn = wrapper!.find('button');
-        refreshBtn.simulate('click');
-        wrapper!.update();
-        expect(errorTagsMock).toBeCalledTimes(2);
-    });
+        const errorState = await screen.findByTestId('docgen-sidebar-error');
+        expect(errorState).toBeInTheDocument();
 
-    test('should render error state', async () => {
-        let wrapper;
-        await act(async () => {
-            wrapper = await getWrapper({
-                ...defaultProps,
-                getDocGenTags: errorTagsMock,
-            });
-        });
-        wrapper!.update();
+        const refreshButton = screen.getByRole('button');
```

Comment:
are we able to add a `name` to this, to be more specific, this will fail if it finds more than one button